IBP-4006 multi app single server impl

grg021 opened the pull request:
added config files for multiple bmsapp instance
added setup script
updated the readme documentation

#IBP-4006

grg021 committed:
Initial impl

grg021 committed:
Update bmsdb env variable

grg021 committed:
Add missing application properties needed by DBScripts

grg021 committed:
create a setup script for proxy and app

grg021 committed:
make setup script executable

grg021 committed:
Update proxy to restart

grg021 committed:
Update readme

grg021 committed:
Update ssl config to restart nginx

grg021 committed:
mount logs folder to current dir

darla-leafnode commented on README.md:
```diff
@@ -2,6 +2,18 @@
+
+## Configurations
+* **.env.example** - this is where we setup environment variables utilized by the docker compose configuration. In order to use it it must be renamed to .env
+* **config** - this contains the configuration files for the BMSScripts inside bmsapp.
```
Change to `DbScripts` (instead of BMSScripts)

grg021 replied:
ok

darla-leafnode commented on README.md:
```diff
@@ -2,6 +2,18 @@
+* **mysql/conf** - this contains mysql configuration that is bind mounted to the mysql container on launch.
+* **nginx** - this contains nginx configuration that is bind mounted to the nginx-proxy container on launch.
+* **docker-compose.yml** - this default configuration is a basic setup of bmsapp with its own database that can be ported to 8080 or 80.
+* **docker-compose-ssl.yml** - this is a setup that uses 4 containers to serve the bmsapp via HTTPS. The TLS Certificate is created and reenwed by the letsencrypt container and the requests are handled by the nginx-proxy container.
```
1. correct typo: **renewed**
2. Let's Encrypt

grg021 replied:
ok thanks

darla-leafnode commented on config/test.properties:
```diff
@@ -0,0 +1,54 @@
+########## Workbench related properties
+########################################
+
+#Workbench hibernate configuration 
```
Why do we need `test.properties` here?

grg021 replied:
removed extra file

darla-leafnode commented on multi-docker-compose-app.yml:
```diff
@@ -0,0 +1,52 @@
+      - BMS_PORT=${BMS_PORT:-443}
+      - BMS_CROPS=${BMS_CROPS:?err}
+      - BMS_DB_HOST=${BMS_DB_HOST:?err}
+      - BMS_DB_PASSWORD=${BMS_DB_PASS:?err}
```
Should this be: **BMS_DB_PASS**=${BMS_DB_PASS:?err}

darla-leafnode commented on docker-compose-ssl.yml:
```diff
@@ -3,6 +3,7 @@ services:
   nginx-proxy:
     image: jwilder/nginx-proxy
     container_name: nginx-proxy
+    restart: always
```
Why do we set to restart = always?

grg021 replied:
For nginx-proxy we need it to always be running. When the host restarts for example we need it also to start automatically.

grg021 replied:
This restarts the nginx-proxy upon start of the server or restart. This is set for all the containers.

darla-leafnode commented on docker-compose-ssl.yml:
```diff
@@ -3,6 +3,7 @@ services:
   nginx-proxy:
```
For bmsapp container, add the BMS_PASSWORD environment property.
Also do the same in docker-compose.yml file

grg021 replied:
done

darla-leafnode commented on .env.sample:
```diff
@@ -1,5 +1,6 @@
 BMS_IMG_URI=ibpbms/bmsapp
 BMS_RELEASE=15.0
```
Change BMS_RELEASE to 16.0

grg021 replied:
done

darla-leafnode commented on setup.sh:
```diff
@@ -0,0 +1,85 @@
+    cp .env.sample .env
+    cp config/application.properties.orig config/application.properties
+
+    sed -i -e "s/BMS_RELEASE=15.0/BMS_RELEASE=$BMS_VERSION/g" .env
```
Change BMS_RELEASE to 16.0

grg021 replied:
done

darla-leafnode reviewed (changes requested):
Please see my comments/questions @grg021

grg021 committed:
update readme

grg021 committed:
carry over changes from multi docker confg

grg021 committed:
Update bms version shown

grg021 committed:
use bmsapp_1 to get more accurate count

grg021 committed:
remove unused config

darla-leafnode commented on docker-compose.yml:
```diff
@@ -3,18 +3,21 @@ services:
       - "80:8080"
     restart: always
     environment:
+      - VIRTUAL_HOST=${BMS_SSL_HOST:-bmsdock.leafnode.io}
```
I think we don't need these environment properties for non-ssl setup:
`VIRTUAL_HOST
VIRTUAL_PORT
LETSENCRYPT_HOST
LETSENCRYPT_EMAIL
BMS_SCHEME
BMS_PORT`
Right?

darla-leafnode commented on docker-compose.yml:
```diff
@@ -26,8 +29,6 @@ services:
     image: mysql:5.6
     ports:
        - "3307:3306"
-    networks:
```
Why did we remove the `networks `config?

grg021 replied:
removed this in the multi container setup but for this config I reverted it back since were not using the default network

darla-leafnode commented on docker-compose.yml:
```diff
@@ -3,18 +3,21 @@ services:
+      - BMS_DB_PASSWORD=${BMS_DB_PASS:?err}
     volumes:
       - mysqldata:/var/lib/mysql
-    links:
```
Why did you remove the `links `config?

grg021 replied:
the links flag is a legacy feature and not required anymore. were already linking them through the network https://docs.docker.com/compose/compose-file/#links

darla-leafnode reviewed (changes requested):
Please check my further comments @grg021

grg021 committed:
updated config file

Added back networks tag since this is not a default network
Removed TLS related config on the bmsapp service

grg021 commented:
Please see updated config file @darla-leafnode
